src/app/api/resident-request/route.ts

```diff
@@ -0,0 +1,36 @@
+import { NextRequest, NextResponse } from "next/server";
+import { prisma } from "lib/prisma";
+import { RequestStatus } from "@prisma/client"; // Import RequestStatus enumD
+
+export async function GET(request: NextRequest) {
+  const { searchParams } = new URL(request.url);
+  const status = searchParams.get("status");
+
+  // Validate the status parameter
+  if (!status) {
+    return NextResponse.json({ error: "Missing status parameter" }, { status: 400 });
+  }
+
+  const validStatuses: RequestStatus[] = ["COMPLETED", "PENDING", "CANCELED"];
```

Comment:
Use the enum instead of typing it out.
```suggestion
  const validStatuses: RequestStatus[] = [RequestStatus.Completed etc...];
```